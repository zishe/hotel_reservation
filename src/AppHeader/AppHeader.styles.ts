import { StyleSheet } from "react-native";

export default StyleSheet.create({
  headerBar: {
    backgroundColor: "#104c97",
    shadowColor: "black",
    shadowOpacity: 1,
    elevation: 4
  },
  profileImgContainer: {
    marginLeft: 5,
    height: 40,
    width: 40,
    borderRadius: 20
  },
  body: {
    flex: 4,
    alignItems: "center"
  },
  titleText: {
    color: "#fff"
  },
  subTitleText: {
    color: "#ddd"
  },
  profileImg: {
    height: 40,
    width: 40,
    borderRadius: 20
  },
  icon: {
    color: "#fff",
    padding: 5
  }
});
