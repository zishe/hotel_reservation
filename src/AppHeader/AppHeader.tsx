import React, { Component } from "react";
import { Header, Left, Title, Right, Text, Body, Icon } from "native-base";
import styles from "./AppHeader.styles";

interface Props {
  title: string;
}

class AppHeader extends Component<Props, {}> {
  render() {
    return (
      <Header
        style={styles.headerBar}
        iosBarStyle="dark-content"
        androidStatusBarColor="lightgray"
      >
        <Left style={{ flex: 2 }}>
          <Icon name="contact" style={styles.icon} />
        </Left>

        <Body style={styles.body}>
          <Title style={styles.titleText}>{this.props.title}</Title>
        </Body>

        <Right style={{ flex: 2 }}>
          <Icon name="menu" style={styles.icon} />
        </Right>
      </Header>
    );
  }
}

export default AppHeader;
