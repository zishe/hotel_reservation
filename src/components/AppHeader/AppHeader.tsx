import React, { Component } from "react";
import { Header, Left, Title, Right, Body, Icon } from "native-base";
import styles from "./styles";

interface Props {
  title: string;
  backScreen?: string;
  navigation?: any;
}

class AppHeader extends Component<Props, {}> {
  render() {
    return (
      <Header
        style={styles.headerBar}
        iosBarStyle="dark-content"
        androidStatusBarColor="lightgray"
      >
        <Left style={{ flex: 2 }}>
          <Icon
            name="menu"
            testID="menu"
            style={styles.icon}
            onPress={this.onOptionsPress}
          />
        </Left>

        <Body style={styles.body}>
          <Title style={styles.titleText}>{this.props.title}</Title>
        </Body>

        <Right style={{ flex: 2 }}>{this.renderRightIcon()}</Right>
      </Header>
    );
  }
  /**
   * @name renderLeftIcon
   * @description render the left-side icon
   */
  renderRightIcon = () => {
    if (this.props.backScreen) {
      return (
        <Icon
          name="close"
          style={styles.icon}
          testID="close"
          onPress={this.onBackPress}
        />
      );
    }

    return <Icon name="contact" testID="contact" style={styles.icon} />;
  };

  /**
   * @name onBackPress
   * @description function to call on press of back icon
   */
  onBackPress = () => {
    this.props.navigation.navigate(this.props.backScreen);
  };

  /**
   * @name onOptionsPress
   * @description function to toggle side menu drawer
   */
  onOptionsPress = () => {
    this.props.navigation.toggleDrawer();
  };
}

export default AppHeader;
