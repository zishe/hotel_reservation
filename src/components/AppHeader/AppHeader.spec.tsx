import React from "react";
import AppHeader from "./AppHeader";
import renderer from "react-test-renderer";

const mockProps = (props: Object) => ({
  title: "Test",
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    toggleDrawer: jest.fn()
  },
  ...props
});

describe.only("Header", () => {
  it("should render as expected", () => {
    const header = renderer.create(<AppHeader title="Test" />).toJSON;
    expect(header).toMatchSnapshot();
  });

  it("should render back icon when navigation prop is provided", () => {
    const props = mockProps({ backScreen: "TestScreen" });
    const header = renderer.create(<AppHeader {...props} />);
    const backIcon = header.root.findAll(
      el => el.type === "Text" && el.props.testID === "close"
    );
    expect(backIcon).toHaveLength(1);
  });

  it("should NOT render user icon when backScreen prop is provided", () => {
    const props = mockProps({ backScreen: "TestScreen" });
    const header = renderer.create(<AppHeader {...props} />);
    const userIcon = header.root.findAll(
      el => el.type === "Text" && el.props.testID === "contact"
    );
    expect(userIcon).toHaveLength(0);
  });

  it("should call navigation.navigate when pressed and some backScreen is provided", () => {
    const props = mockProps({ backScreen: "TestScreen" });
    const header = renderer.create(<AppHeader {...props} />);
    const backIcon = header.root.find(
      el => el.type === "Text" && el.props.testID === "close"
    );
    backIcon.props.onPress();
    expect(props.navigation.navigate).toHaveBeenCalledWith(
      (props as any).backScreen
    );
  });

  it("should call navigation.toggleDrawer when menu icon is pressed", () => {
    const props = mockProps({});
    const header = renderer.create(<AppHeader {...props} />);
    const backIcon = header.root.find(
      el => el.type === "Text" && el.props.testID === "menu"
    );
    backIcon.props.onPress();
    expect(props.navigation.toggleDrawer).toBeCalled();
  });
});
