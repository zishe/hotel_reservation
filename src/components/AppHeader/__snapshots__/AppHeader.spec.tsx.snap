// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render as expected 1`] = `[Function]`;
